sdk/Sdk/worker.config.json

```diff
@@ -3,6 +3,7 @@
     "language": "dotnet-isolated",
     "extensions": [ ".dll" ],
     "defaultExecutablePath": "$functionExe$",
-    "defaultWorkerPath": "$functionWorker$"
+    "defaultWorkerPath": "$functionWorker$",
+    "workerIndexing":  "true"
```

Comment:
**General idea of new design:** When a worker.config file has "workerIndexing" set to true, it means that the worker knows how to respond to a metadata request from the host. The worker may decide for itself after getting the request whether it will index the functions and return the metadata or force the indexing responsibility back onto the host. 

The design hasn't been outlined yet or implemented in the host (still under discussion) but wanted to provide context for this.

cc: @fabiocav @surgupta-msft